sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -146,6 +146,7 @@ abstract class ExternalCatalog {
 
   def createFunction(db: String, funcDefinition: CatalogFunction): Unit
 
+  // TODO: dropFunction should take a FunctionIdentifier.
```

Comment:
I don't think so... it's largely by design that we just pass in the strings here. This is the external catalog which shouldn't know anything about current databases, so the effective database should already be resolved at this point.